clientconn.go

```diff
@@ -119,6 +132,7 @@ type ClientConn struct {
 
 func (cc *ClientConn) resetTransport(closeTransport bool) error {
 	var retries int
+	totime := time.Now().Add(cc.dopts.timeout)
```

Comment:
Cool, that reads much better. How did you feel about checking if the ensuing time.Sleep(dur) call would take us past the timeout and, in that case, stopping prematurely?